kartothek/serialization/_generic.py

```diff
@@ -203,6 +203,49 @@ def filter_df_from_predicates(df, predicates, strict_date_types=False):
     return df[indexer]
 
 
+def _ensure_type_stability(array_like, value, strict_date_types):
```

Comment:
please add a docstring describing this function. Even though it is an internal function, it is quite complex and devs should be able to understand it easiliy